qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/web/views/create/offer/CustomerSelectionView.groovy

```diff
@@ -113,11 +123,14 @@ class CustomerSelectionView extends VerticalLayout{
      */
     private def generateCustomerGrid() {
         try {
-            this.customerGrid.addColumn({ customer -> customer.title }).setCaption("Title")
-            this.customerGrid.addColumn({ customer -> customer.firstName }).setCaption("First Name")
-            this.customerGrid.addColumn({ customer -> customer.lastName }).setCaption("Last Name")
-            this.customerGrid.addColumn({ customer -> customer.emailAddress }).setCaption("Email Address")
-            //this.customerGrid.addColumn({ customer -> customer.getAffiliations().toString() }).setCaption("Affiliation")
+            this.customerGrid.addColumn({ customer -> customer.title })
+                    .setCaption("Title").setId("title")
+            this.customerGrid.addColumn({ customer -> customer.firstName })
+                    .setCaption("First Name").setId("FirstName")
+            this.customerGrid.addColumn({ customer -> customer.lastName })
+                    .setCaption("Last Name").setId("LastName")
+            this.customerGrid.addColumn({ customer -> customer.emailAddress })
+                    .setCaption("Email Address").setId("EmailAddress")
```

Comment:
```suggestion        
        ListDataProvider<Customer> customerDataProvider = new ListDataProvider(searchCustomerViewModel.foundCustomers)
        grid.setDataProvider(customerDataProvider)

        HeaderRow customerFilterRow = grid.appendHeaderRow()

        Grid.Column<Customer, String> titleColumn = grid.addColumn({ customer -> customer.title.value }).setCaption("Title")
        GridUtils.setupColumnFilter(customerDataProvider, titleColumn, customerFilterRow)

        Grid.Column<Customer, String> firstNameColumn = grid.addColumn({ customer -> customer.getFirstName() }).setCaption("First Name")
        GridUtils.setupColumnFilter(customerDataProvider, firstNameColumn, customerFilterRow)

        Grid.Column<Customer, String> lastNameColumn = grid.addColumn({ customer -> customer.getLastName() }).setCaption("Last Name")
        GridUtils.setupColumnFilter(customerDataProvider, lastNameColumn, customerFilterRow)

        Grid.Column<Customer, String> emailColumn = grid.addColumn({ customer -> customer.getEmailAddress() }).setCaption("Email Address")
        GridUtils.setupColumnFilter(customerDataProvider, emailColumn, customerFilterRow)

Reply:
This way the filters, data providers etc are only locally accessible and bound to a variable. As they should not be changed later on this is desired I think.

Reply:
The `foundCustomerList` should not be taken as data provider since it is not changed or observable and no list from the view itself is intended to be used here.

Reply:
> This way the filters, data providers etc are only locally accessible and bound to a variable. As they should not be changed later on this is desired I think.

As you can access the `projectManager` grid because its modifier is public, nothing is "local" atm, my implementation has not changed the scope, nore does it expose anything in addition than before.

Reply:
> The `foundCustomerList` should not be taken as data provider since it is not changed or observable and no list from the view itself is intended to be used here. (please move to the view model for data providing)

This was not introduced in this PR, so I would argue this is a topic for a separate issue and PR to work on it.

Reply:
> > The `foundCustomerList` should not be taken as data provider since it is not changed or observable and no list from the view itself is intended to be used here. (please move to the view model for data providing)
> 
> This was not introduced in this PR, so I would argue this is a topic for a separate issue and PR to work on it.

Since you modify the data provider in this PR I would argue that this is indeed part of this PR. Removing the old list might be different but using a list from the correct location for your newly setup data provider can be viewed of being part of this PR.